### ISSUE TITLE
Android workspace preview is not available

### ISSUE THREAD

**Issue by mindgsv** [codenvy/codenvy#2092]
When I started fresh new Android stack and run example app build everything went fine. I was able to click on provided preview link in the terminal and accesses VNC window with android emulator which was starting like 15 minutes and I waited... After about 10-15 minutes when android emulator booted up my workspace timedout and was stoped, because I was using FREE account just to checkout the service. However after automatic workspace stop I was not able to access VNC window again it was always "Failed to connect" or something like this always the only thing which fixed that was workspace deletation and creation of new.
Reproduce:
1. Start new android stack workspace
2. Run android example app 
3. Open VNC preview link in the terminal (which is given on app build)
4.  Let the workspace to timeout
5. Start workspace after timeout
6. Run app build, try to open VNC window and now "Failed to connect" should be present

Besides android emulation speed is far for being usable or where is a reason for that?

**Comment by eivantsov** [comment]
@mindgsv when a workspace restarts you should re-run run command which will generate a new preview URL.

As to Android emulation speed, indeed it takes a while to boot with default 2GB per workspace. Have you tried allocating 3GB?

**Comment by mindgsv** [comment]
@eivantsov I have tried with re run, but problem persisted even after workspace restart with 3GB's
However I will try to reproduce it and let you know.

**Comment by eivantsov** [comment]
What exactly do you mean by `problem persisted`? Slow Emulator start?

**Comment by mindgsv** [comment]
I tried again today with 3GB RAM allocated and it work as expected.
The problem for me probably was some glitch, probably I have pressed few times "Emulator" in VNC window which resulted in starting 2 or even more android emulator instances which was the cause of the problem with slow emulator. As user I have no idea in those several seconds while nothing happen after pressing "Emulator" so I might do it again few times and emulator let's you start as many instances as you like :)
However everything is OK now. Thanks for the help @eivantsov 👍
